src/JitInterface/src/JitConfigProvider.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.JitInterface
+{
+    internal sealed class JitConfigProvider
+    {
+        private Dictionary<string, string> _config = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private object _keepAlive; // Keeps callback delegates alive
+
+        public IntPtr UnmanagedInstance
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="JitConfigProvider"/>.
+        /// </summary>
+        /// <param name="parameters">Name-value pairs separated by an equals sign.</param>
+        public JitConfigProvider(IEnumerable<string> parameters)
+        {
+            foreach (var param in parameters)
+            {
+                int indexOfEquals = param.IndexOf('=');
+
+                // We're skipping bad parameters without reporting.
+                // This is not a mainstream feature that would need to be friendly.
+                // Besides, to really validate this, we would also need to check that the config name is known.
+                if (indexOfEquals < 1)
+                    continue;
+
+                string name = param.Substring(0, indexOfEquals);
+                string value = param.Substring(indexOfEquals + 1);
+
+                _config[name] = value;
+            }
+
+            UnmanagedInstance = CreateUnmanagedInstance();
+        }
+
+        public int GetIntConfigValue(string name, int defaultValue)
+        {
+            string stringValue;
+            int intValue;
+            if (_config.TryGetValue(name, out stringValue) &&
+                Int32.TryParse(stringValue, out intValue))
```

Comment:
Do you mean this should always be considered a hex number? `AllowHexSpecifier` is an odd option because it always leads me to think that I can optionally prepend `0x` to make it hex but that's not how it works - so just double checking. I can't for my life find how this is parsed in CoreCLR.